modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/NativeMediaManager.java

```diff
@@ -104,15 +105,18 @@ protected NativeMediaManager() {
          */
         try {
             AccessController.doPrivileged((PrivilegedExceptionAction) () -> {
+                ArrayList<String> dependencies = new ArrayList<>();
                 if (HostUtils.isWindows() || HostUtils.isMacOSX()) {
                     NativeLibLoader.loadLibrary("glib-lite");
                 }
 
                 if (!HostUtils.isLinux() && !HostUtils.isIOS()) {
                     NativeLibLoader.loadLibrary("gstreamer-lite");
+                } else {
+                    dependencies.add("gstreamer-lite");
```

Comment:
For Linux, the following are also needed as dependencies:

```
                    if (HostUtils.isLinux()) {
                        dependencies.add("fxplugins");
                        dependencies.add("avplugin-54");
                        dependencies.add("avplugin-56");
                        dependencies.add("avplugin-57");
                        dependencies.add("avplugin-ffmpeg-56");
                        dependencies.add("avplugin-ffmpeg-57");
                    }
```

With the addition of the above, I can run media (although I get warnings...see my comment in NativeLibLoader).

Reply:
I don't see them in NativeLibLoader.java. What warnings would that be?

Reply:
I added a boolean "load" to the loadLibraryFromResource method, which is true by default, but false when loading dependencies. In that case, the resource is simply unpacked.